metadetect/metadetect.py

```diff
@@ -222,82 +234,214 @@ def go(self):
                 if np.all((obs.bmask & self['nodet_flags']) != 0):
                     any_all_masked = True
 
-        if not np.any(self.mfrac < 1) or any_all_zero_weight or any_all_masked:
+        if not np.any(mfrac < 1) or any_all_zero_weight or any_all_masked:
```

Comment:
we need all bands, so should we not be demanding `np.all(mfrac < 1)`?

Reply:
```suggestion
        # if the there are no pixels with mfrac < 1 or it is all zero weight
        # or it is all masked, we cannot measure anything so set result to None
        if (not np.any(mfrac < 1)) or any_all_zero_weight or any_all_masked:
```
mfrac is summed over all bands already and is an array of size the size of the image. We are checking if it has any entries less than 1. If it does not, we set the result to None. I added extra parens to try and a comment clear up the logic.